xbmc/video/VideoInfoTag.cpp

```diff
@@ -820,3 +869,168 @@ unsigned int CVideoInfoTag::GetDurationFromMinuteString(const std::string &runti
+{
+  m_country = Trim(std::move(country));
+}
+
+void CVideoInfoTag::SetTagLine(const std::string &tagLine)
+{
+  m_strTagLine = Trim(tagLine);
+}
+
+void CVideoInfoTag::SetPlotOutline(const std::string &plotOutline)
+{
+  m_strPlotOutline = Trim(plotOutline);
+}
+
+void CVideoInfoTag::SetTrailer(const std::string &trailer)
+{
+  m_strTrailer = Trim(trailer);
+}
+
+void CVideoInfoTag::SetPlot(const std::string &plot)
+{
+  m_strPlot = Trim(plot);
+}
+
+void CVideoInfoTag::SetTitle(const std::string &title)
+{
+  m_strTitle = Trim(title);
+}
+
+void CVideoInfoTag::SetSortTitle(const std::string &sortTitle)
+{
+  m_strSortTitle = Trim(sortTitle);
+}
+
+void CVideoInfoTag::SetPictureURL(const CScraperUrl &pictureURL)
+{
+  m_strPictureURL = pictureURL;
+}
+
+void CVideoInfoTag::SetVotes(const std::string &votes)
+{
+  m_strVotes = Trim(votes);
+}
+
+void CVideoInfoTag::SetArtist(const std::vector<std::string> &artist)
+{
+  m_artist = Trim(std::move(artist));
+}
+
+void CVideoInfoTag::SetSet(const std::string &set)
+{
+  m_strSet = Trim(set);
+}
+
+void CVideoInfoTag::SetTags(const std::vector<std::string> &tags)
+{
+  m_tags = Trim(std::move(tags));
+}
+
+void CVideoInfoTag::SetFile(const std::string &file)
+{
+  m_strFile = Trim(file);
+}
+
+void CVideoInfoTag::SetPath(const std::string &path)
+{
+  m_strPath = Trim(path);
+}
+
+void CVideoInfoTag::SetIMDBNumber(const std::string &imdbNumber)
+{
+  m_strIMDBNumber = Trim(imdbNumber);
+}
+
+void CVideoInfoTag::SetMPAARating(const std::string &mpaaRating)
+{
+  m_strMPAARating = Trim(mpaaRating);
+}
+
+void CVideoInfoTag::SetFileNameAndPath(const std::string &fileNameAndPath)
+{
+  m_strFileNameAndPath = Trim(fileNameAndPath);
+}
+
+void CVideoInfoTag::SetOriginalTitle(const std::string &originalTitle)
+{
+  m_strOriginalTitle = Trim(originalTitle);
+}
+
+void CVideoInfoTag::SetEpisodeGuide(const std::string &episodeGuide)
+{
+  m_strEpisodeGuide = Trim(episodeGuide);
+}
+
+void CVideoInfoTag::SetStatus(const std::string &status)
+{
+  m_strStatus = Trim(status);
+}
+
+void CVideoInfoTag::SetProductionCode(const std::string &productionCode)
+{
+  m_strProductionCode = Trim(productionCode);
+}
+
+void CVideoInfoTag::SetShowTitle(const std::string &showTitle)
+{
+  m_strShowTitle = Trim(showTitle);
+}
+
+void CVideoInfoTag::SetStudio(const std::vector<std::string> &studio)
+{
+  m_studio = Trim(std::move(studio));
+}
+
+void CVideoInfoTag::SetAlbum(const std::string &album)
+{
+  m_strAlbum = Trim(album);
+}
+
+void CVideoInfoTag::SetShowLink(const std::vector<std::string> &showLink)
+{
+  m_showLink = Trim(std::move(showLink));
+}
+
+void CVideoInfoTag::SetUniqueId(const std::string &uniqueId)
+{
+  m_strUniqueId = Trim(uniqueId);
+}
+
+std::string CVideoInfoTag::Trim(const std::string &value)
+{
+  std::string v(value);
+  return StringUtils::Trim(v);
+}
+
+std::vector<std::string> CVideoInfoTag::Trim(const std::vector<std::string>&& items)
```

Comment:
Remove const and remove the assignment to values below, we want to get rid of copying and use move semantics as much as possible